code/modules/antagonists/heretic/knowledge/rust_lore.dm

```diff
@@ -265,10 +291,44 @@
 		sound = 'sound/ambience/antag/heretic/ascend_rust.ogg',
 		color_override = "pink",
 	)
-	new /datum/rust_spread(loc)
+	//new /datum/rust_spread(loc)
+	trigger(loc)
 	RegisterSignal(user, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
 	RegisterSignal(user, COMSIG_LIVING_LIFE, PROC_REF(on_life))
 	user.client?.give_award(/datum/award/achievement/misc/rust_ascension, user)
+	var/datum/action/cooldown/spell/aoe/rust_conversion/rust_spread_spell = locate() in user.actions
+	rust_spread_spell?.cooldown_time /= 2
+
+// I sure hope this doesn't have performance implications
+/datum/heretic_knowledge/ultimate/rust_final/proc/trigger(turf/center)
+	var/greatest_dist = 0
+	var/list/turfs_to_transform = list()
+	for (var/turf/transform_turf as anything in GLOB.station_turfs)
+		if (transform_turf.turf_flags & NO_RUST)
+			continue
+		var/dist = get_dist(center, transform_turf)
+		if (dist > greatest_dist)
+			greatest_dist = dist
+		if (!turfs_to_transform["[dist]"])
+			turfs_to_transform["[dist]"] = list()
+		turfs_to_transform["[dist]"] += transform_turf
+
+	for (var/iterator in 1 to greatest_dist)
+		if(!turfs_to_transform["[iterator]"])
+			continue
+		addtimer(CALLBACK(src, PROC_REF(transform_area), turfs_to_transform["[iterator]"]), (5 SECONDS) * iterator)
+
+/datum/heretic_knowledge/ultimate/rust_final/proc/transform_area(list/turfs)
+	turfs = shuffle(turfs)
+	var/first_third = turfs.Copy(1, round(length(turfs) * 0.33))
+	var/second_third = turfs.Copy(round(length(turfs) * 0.33), round(length(turfs) * 0.66))
+	var/third_third = turfs.Copy(round(length(turfs) * 0.6), length(turfs))
+	for(var/turf/turf as anything in first_third)
+		turf.rust_heretic_act(5)
+	for(var/turf/turf as anything in second_third)
+		addtimer(CALLBACK(turf, TYPE_PROC_REF(/atom, rust_heretic_act), 5), 5 SECONDS * 0.33)
+	for(var/turf/turf as anything in third_third)
+		addtimer(CALLBACK(turf, TYPE_PROC_REF(/atom, rust_heretic_act), 5), 5 SECONDS * 0.66)
```

Comment:
This is an insane amount of timer spam, Like up to 500 timers or w/e 

Rather than making 500 times to transform every single turf, just have 1 timer to transform all the turfs, IE via another proc
```dm
	addtimer(CALLBACK(src, PROC_REF(delay_transform_turfs)), 5 SECONDS * 0.33)
```
```dm
/datum/heretic_knowledge/ultimate/rust_final/proc/delay_transform_turfs(list/turfs)
	for(var/turf/turf as anything in first_third)
		turf.rust_heretic_act(5)
		CHECK_TICK
```